packages/web/components/Dashboard/Calendar.tsx

```diff
@@ -0,0 +1,645 @@
+import {
+  Box,
+  ButtonGroup,
+  CalendarIcon,
+  ExternalLinkIcon,
+  IconButton,
+  Image,
+  Popover,
+  PopoverBody,
+  PopoverCloseButton,
+  PopoverContent,
+  PopoverFooter,
+  PopoverHeader,
+  PopoverTrigger,
+  Portal,
+  VStack,
+} from '@metafam/ds';
+import React, { useEffect, useState } from 'react';
+
+export const Calendar: React.FC = () => {
+  //   const [result, setResult] = useState([]);
+  const [loading, setLoading] = useState(false);
+
+  useEffect(() => {
+    async function getData() {
+      try {
+        setLoading(true);
+
+        // const settings = {
+        //   method: 'POST',
+        //   headers: {
+        //     Accept: 'application/json',
+        //     'Content-Type': 'application/json',
+        //   },
+        // };
+
+        // const cal_id = 'nih59ktgafmm64ed4qk6ue8vv4@group.calendar.google.com';
+
+        // const api_key = 'AIzaSyCdI6EdlH7x65i_CH7RmPl1AJ-UKarujQo';
+
+        // const fetchResponse = await fetch(
+        //   //   `https://www.googleapis.com/calendar/v3/calendars/${cal_id}/events?key=${api_key}`,
+        //   'https://www.googleapis.com/calendar/v3/calendars/nih59ktgafmm64ed4qk6ue8vv4@group.calendar.google.com/events/import',
+        //   settings,
+        // );
+        // const data = await fetchResponse.json();
+        // setResult(data);
+        setLoading(false);
+      } catch (error) {
+        setLoading(false);
+      }
+    }
+
+    getData();
+  }, []);
+
+  return (
+    <div>
+      {loading ? (
+        <div>loading</div>
+      ) : (
+        <VStack
```

Comment:
hey @dysbulic, there has been a slight misunderstanding, because the branch you're currently viewing is implementing only the xp section and because it hasn't been rebased to the latest develop branch, it's also showing changes for other sections (which are basically just mock data). The implementation of calendar section, which you reviewed, is done here: https://github.com/MetaFam/TheGame/pull/902 and has a very similar structure as you suggested.